File: 0049-group-anagrams/0049-group-anagrams.java
class Solution {
    public List<List<String>> groupAnagrams(String[] str) {
        
        int n= str.length; 
        Map<String, List<String>> map = new HashMap<>();


        for(int i=0;i<n; i++){
            String word= str[i]; // eat
            
            char[] ch= word.toCharArray();  //'e' 'a' 't'

            Arrays.sort(ch); //  a e t 

            String sortedword= new String(ch); 
            
            if(!map.containsKey(sortedword))
                map.put(sortedword, new ArrayList<>()); 
            
            map.get(sortedword).add(word); 
        }

        return new ArrayList<>(map.values()); 
        
    }
}

File: 0049-group-anagrams/0049-group-anagrams.py
class Solution(object):
    def groupAnagrams(self, strs):
        """
        :type strs: List[str]
        :rtype: List[List[str]]
        """
        dict={}
        for s in strs:  # s= eat 
            sort= ''.join(sorted(s))

            if sort in dict: 
                dict[sort].append(s)
            else : 
                dict[sort]= [s] 

        return dict.values()